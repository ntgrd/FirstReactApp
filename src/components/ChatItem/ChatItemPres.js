import {IconButton, ListItem, ListItemAvatar} from "@material-ui/core";
import Avatar from "@material-ui/core/Avatar";
import DeleteIcon from '@material-ui/icons/Delete';
import List from "@material-ui/core/List";
import {makeStyles} from "@material-ui/core/styles";
import {Link} from "react-router-dom";
import './ChatItem.css'

const useStyles = makeStyles((theme) => ({
    root: {
        '& > *': {
            margin: theme.spacing(1),
        },
        width: '100%',
        maxWidth: '36ch',
        backgroundColor: theme.palette.background.paper,
    },
    inline: {
        display: 'inline',
    },
}));

export const ChatItemPres = ({id, chatName, onDelete, current}) => {
    const classes = useStyles();

    return (
        <List className={classes.root}>
            <ListItem className='chat_item' key={id} alignItems="flex-start" divider={true}>
                <ListItemAvatar>
                    <Avatar alt="Remy Sharp" src="/static/images/avatar/1.jpg" data-testid="avatar"/>
                </ListItemAvatar>
                {/*<Link to={`/chats/${id}`}>*/}
                {/*    <b style={{color: current ? '#000000' : 'grey'}}>*/}
                {/*        <h3>{chatName}</h3>*/}
                {/*    </b>*/}
                {/*</Link>*/}
                <IconButton aria-label="delete" onClick={() => onDelete(id)}>
                    <DeleteIcon/>
                </IconButton>
            </ListItem>
        </List>
    );
};
