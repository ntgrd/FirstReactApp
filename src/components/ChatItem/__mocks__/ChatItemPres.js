import React from "react";

const ChatItemPres = () => (
        <div data-testid="ChatItemPres"/>
    )
export default ChatItemPres;
