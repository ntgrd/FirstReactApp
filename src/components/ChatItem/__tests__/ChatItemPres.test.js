import React from 'react';
import {render, screen} from '@testing-library/react';
import {ChatItemPres} from "../ChatItemPres";
import userEvent from '@testing-library/user-event';

describe('ChatItemPres', () => {
    it('matches snapshot', () => {
        const component = render(
            <ChatItemPres/>
        );

        expect(component).toMatchSnapshot();
    });

    it("render List", () => {
        render(
            <ChatItemPres/>
        );

        expect(screen.getByRole('list')).toBeInTheDocument();
        expect(screen.getByRole('listitem')).toBeInTheDocument();
        expect(screen.getByTestId('avatar')).toBeInTheDocument()
    });

    it("button Delete works", () => {
        const onDelete = jest.fn();

        render(
            <ChatItemPres onDelete={onDelete}/>
        );
        const button = screen.getByRole('button');
        userEvent.click(button);

        expect(onDelete.mock.calls.length).toBe(1);
    });
})
