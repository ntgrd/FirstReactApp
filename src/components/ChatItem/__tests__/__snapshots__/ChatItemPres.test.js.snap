// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatItemPres matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="MuiList-root makeStyles-root-1 MuiList-padding"
      >
        <li
          class="MuiListItem-root chat_item MuiListItem-gutters MuiListItem-divider MuiListItem-alignItemsFlexStart"
        >
          <div
            class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular"
              data-testid="avatar"
            >
              <img
                alt="Remy Sharp"
                class="MuiAvatar-img"
                src="/static/images/avatar/1.jpg"
              />
            </div>
          </div>
          <button
            aria-label="delete"
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="MuiList-root makeStyles-root-1 MuiList-padding"
    >
      <li
        class="MuiListItem-root chat_item MuiListItem-gutters MuiListItem-divider MuiListItem-alignItemsFlexStart"
      >
        <div
          class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular"
            data-testid="avatar"
          >
            <img
              alt="Remy Sharp"
              class="MuiAvatar-img"
              src="/static/images/avatar/1.jpg"
            />
          </div>
        </div>
        <button
          aria-label="delete"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
